--- goo/gfile.cc	Mon Feb 18 19:24:19 2019
+++ goo/gfile.cc	Wed Mar 13 13:33:37 2019
@@ -602,7 +602,7 @@
       }
     }
     wPath[i] = (wchar_t)0;
-    for (i = 0; mode[i] && i < sizeof(wMode) - 1; ++i) {
+    for (i = 0; mode[i] && i < (sizeof(wMode) / sizeof(wchar_t)) - 1; ++i) {
       wMode[i] = (wchar_t)(mode[i] & 0xff);
     }
     wMode[i] = (wchar_t)0;
--- xpdf/PDFDoc.cc	Mon Feb 18 19:24:19 2019
+++ xpdf/PDFDoc.cc	Wed Mar 13 13:36:11 2019
@@ -223,6 +223,7 @@
     fileNameU = NULL;
 #endif
   }
+  str = strA;
   ok = setup(ownerPassword, userPassword);
 }
 
--- xpdf/XFAForm.cc	Mon Feb 18 19:24:19 2019
+++ xpdf/XFAForm.cc	Wed Mar 13 13:37:10 2019
@@ -2732,7 +2732,7 @@
 	      u <<= 4;
 	      if (c >= '0' && c <= '9') {
 		u += c - '0';
-	      } else if (c >= 'a' && c <= 'F') {
+	      } else if (c >= 'a' && c <= 'f') {
 		u += c - 'a' + 10;
 	      } else if (c >= 'A' && c <= 'F') {
 		u += c - 'A' + 10;
@@ -2929,7 +2929,7 @@
 	      u <<= 4;
 	      if (c >= '0' && c <= '9') {
 		u += c - '0';
-	      } else if (c >= 'a' && c <= 'F') {
+	      } else if (c >= 'a' && c <= 'f') {
 		u += c - 'a' + 10;
 	      } else if (c >= 'A' && c <= 'F') {
 		u += c - 'A' + 10;
@@ -3104,7 +3104,7 @@
 	      u <<= 4;
 	      if (c >= '0' && c <= '9') {
 		u += c - '0';
-	      } else if (c >= 'a' && c <= 'F') {
+	      } else if (c >= 'a' && c <= 'f') {
 		u += c - 'a' + 10;
 	      } else if (c >= 'A' && c <= 'F') {
 		u += c - 'A' + 10;
--- xpdf/XRef.cc	Mon Feb 18 19:24:19 2019
+++ xpdf/XRef.cc	Wed Mar 13 13:37:38 2019
@@ -1152,15 +1152,15 @@
 #if MULTITHREADED
   gLockMutex(&objStrsMutex);
 #endif
-  if (!(objStr = getObjectStream(objStrNum))) {
-    return gFalse;
-  }
+  objStr = getObjectStream(objStrNum);
+  if (objStr) {
   cleanObjectStreamCache();
   objStr->getObject(objIdx, objNum, obj);
+  }
 #if MULTITHREADED
   gUnlockMutex(&objStrsMutex);
 #endif
-  return gTrue;
+  return objStr ? gTrue : gFalse;
 }
 
 // NB: objStrsMutex must be locked when calling this function.
