--- xpdf/GfxFont.cc	Mon Feb 18 19:24:19 2019
+++ xpdf/GfxFont.cc	Mon Mar 11 13:48:26 2019
@@ -820,7 +820,7 @@
   fontLoc->oblique = oblique;
   return fontLoc;
 }
-
+#define GFX_BLOCK_SIZE 64*1024
 char *GfxFont::readEmbFontFile(XRef *xref, int *len) {
   char *buf;
   Object obj1, obj2;
@@ -842,14 +842,14 @@
   buf = NULL;
   str->reset();
   do {
-    if (size > INT_MAX - 4096) {
+    if (size > INT_MAX - GFX_BLOCK_SIZE) {
       error(errSyntaxError, -1, "Embedded font file is too large");
       break;
     }
-    buf = (char *)grealloc(buf, size + 4096);
-    n = str->getBlock(buf + size, 4096);
+    buf = (char *)grealloc(buf, size + GFX_BLOCK_SIZE);
+    n = str->getBlock(buf + size, GFX_BLOCK_SIZE);
     size += n;
-  } while (n == 4096);
+  } while (n == GFX_BLOCK_SIZE);
   *len = size;
   str->close();
 
--- xpdf/XRef.h	Mon Feb 18 19:24:19 2019
+++ xpdf/XRef.h	Fri Mar  8 14:31:38 2019
@@ -50,7 +50,7 @@
   Object obj;
 };
 
-#define xrefCacheSize 16
+#define xrefCacheSize 32
 
 #define objStrCacheSize 128
 #define objStrCacheTimeout 1000
